react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+let counter = 0;
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      numberClicks: counter
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState({
+      numberClicks: counter++
```

Comment:
You can access the previous count at `this.state.numberClicks` instead of introducing a whole new variable.